src/NServiceBus.AzureStoragePersistence.AcceptanceTests/Sagas/When_saga_is_started_by_two_types_of_messages.cs

```diff
@@ -0,0 +1,163 @@
+        }
+
+        private static IEnumerable<int> AllIndexesOf(string str, string value)
+        {
+            if (str == null || value == null)
+            {
+                yield break;
+            }
+
+            for (var i = 0;; i += value.Length)
+            {
+                i = str.IndexOf(value, i, StringComparison.Ordinal);
+                if (i == -1)
+                {
+                    yield break;
+                }
+                yield return i;
+            }
+        }
+
+        public class ProvideConfiguration : IProvideConfiguration<TransportConfig>
+        {
+            public TransportConfig GetConfiguration()
+            {
+                return new TransportConfig
+                {
+                    MaximumConcurrencyLevel = 3
+                };
+            }
+        }
+
+        public class Context : ScenarioContext
+        {
+            private readonly ConcurrentDictionary<string, string> completed = new ConcurrentDictionary<string, string>();
+            public int CompletedIdsCount => completed.Count;
+            public IEnumerable<string> CompletedIds => completed.Keys;
+
+            public void MarkAsCompleted(string orderId)
+            {
+                completed.AddOrUpdate(orderId, orderId, (o1, o2) => o1);
+            }
+        }
+
+        public class ReceiverWithSagas : EndpointConfigurationBuilder
+        {
+            public ReceiverWithSagas()
+            {
+                EndpointSetup<DefaultServer>(
+                    config => { });
+            }
+        }
+
+        public class ShippingPolicy : Saga<ShippingPolicy.State>,
+            IAmStartedByMessages<OrderPlaced>,
+            IAmStartedByMessages<OrderBilled>
+        {
+            public Context Context { get; set; }
+
+            public void Handle(OrderBilled message)
+            {
+                Data.OrderId = message.OrderId;
+                Data.Billed = true;
+
+                TryComplete();
+            }
+
+            public void Handle(OrderPlaced message)
+            {
+                Data.OrderId = message.OrderId;
+                Data.Placed = true;
+
+                TryComplete();
+            }
+
+            private void TryComplete()
+            {
+                if (Data.Billed && Data.Placed)
+                {
+                    MarkAsComplete();
+                    Context.MarkAsCompleted(Data.OrderId);
+                }
+            }
+
+            protected override void ConfigureHowToFindSaga(SagaPropertyMapper<State> mapper)
+            {
+                mapper.ConfigureMapping<OrderPlaced>(m => m.OrderId)
+                    .ToSaga(s => s.OrderId);
+                mapper.ConfigureMapping<OrderBilled>(m => m.OrderId)
+                    .ToSaga(s => s.OrderId);
+            }
+
+            public class State : IContainSagaData
+            {
+                [Unique]
```

Comment:
Yes. I'll make sure, it's there. Should this be a part of this PR, or should we proceed with release notes first and then PR docos?